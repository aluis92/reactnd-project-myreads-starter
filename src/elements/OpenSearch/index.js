import React from 'react';
import { string } from 'prop-types';

import {
    Container,
    OpenSearchLink,
} from './styles';

function OpenSearch({
    text,
    to,
}) {
    return (
        <Container>
            <OpenSearchLink to={to}>{text}</OpenSearchLink>
        </Container>
    );
}

OpenSearch.propTypes = {
    text: string.isRequired,
    to: string.isRequired,
};

export default OpenSearch;
