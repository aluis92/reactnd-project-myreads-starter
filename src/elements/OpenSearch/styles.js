import styled from 'styled-components';
import { Link } from 'react-router-dom';
import addIcon from '../../assets/icons/add.svg';

export const Container = styled.header`
    position: fixed;
    right: 25px;
    bottom: 25px;
`;

export const OpenSearchLink = styled(Link)`
    display: block;
    width: 50px;
    height: 50px;
    border-radius: 50%;
    background: #2e7d32;
    background-image: url(${addIcon});
    background-repeat: no-repeat;
    background-position: center;
    background-size: 28px;
    box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
    font-size: 0;
`;
