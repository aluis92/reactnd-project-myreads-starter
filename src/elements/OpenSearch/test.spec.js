import React from 'react';
import {
    shallow,
} from 'enzyme';

import OpenSearch from './index';

describe('OpenSearch component', () => {
    const props = {
        text: 'ok',
        to: '/',
    };

    it('should render if it has text and to props', () => {
        expect(shallow(
            <OpenSearch
                {...props}
            />,
        )).toMatchSnapshot();
    });
});
