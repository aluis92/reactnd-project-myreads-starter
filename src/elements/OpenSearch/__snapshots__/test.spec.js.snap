// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenSearch component should render if it has text and to props 1`] = `
<styled.header>
  <Styled(Link)
    to="/"
  >
    ok
  </Styled(Link)>
</styled.header>
`;
